tests/test_flow/generate_test_compatibility_cache.py

```diff
@@ -40,40 +40,48 @@ def __init__(self, cache_dir, make_counter):
         xy_counter = make_counter()
 
         @builder
-        @xy_counter
         def xy(x, y):
+            xy_counter.mark()
             return x * y
 
         yz_counter = make_counter()
 
         @builder
-        @yz_counter
         def yz(y, z):
+            yz_counter.mark()
             return y * z
 
         xy_plus_yz_counter = make_counter()
 
         @builder
-        @xy_plus_yz_counter
         def xy_plus_yz(xy, yz):
+            xy_plus_yz_counter.mark()
             return xy + yz
 
-        self.flow = builder.build()
         self.xy_counter = xy_counter
         self.yz_counter = yz_counter
         self.xy_plus_yz_counter = xy_plus_yz_counter
 
+        self.manual_flow = builder.build()
+        builder.set("core__versioning_mode", "auto")
```

Comment:
Wondering, does the `auto` and `manual` modes make a difference to the tests here?

Reply:
A flow with `auto` mode will also add the bytecode of the function to its functional hash. Whereas manual mode wouldn't. Adding this test for auto mode will check for cache compatibility for auto versioning mode as well. Similar to `manual` mode cache compatibility check, the detection of bytecode hashing logic change is not exhaustive. But it should at least catch major changes to the hashing logic.